wp/lib/bap_wp/src/compare.mli

```diff
@@ -104,6 +104,20 @@ val compare_subs_empty_post : comparator * comparator
     creation time. *)
 val compare_subs_sp : comparator * comparator
 
+(** Compare two subs by composition for restraints on a pointer.:
```

Comment:
I don't think this is an appropriate description of this function: nothing in the body of the function mentions pointers in any way.

In general this function seems very generic, so I'm not sure why you're mentioning x86_64, or indeed why the description is what it is. It looks like this function just bottles up pre and post conditions. The comment should reflect that.